VertexAISnippets/VertexAISnippets/VertexAISnippets.swift

```diff
@@ -0,0 +1,329 @@
+    if let text = response.text {
+      print(text)
+    }
+    // [END two_image_prompt]
+  }
+
+  func textAndVideoPrompt() async throws {
+    // AVFoundation support coming soon™
+    // [START text_video_prompt]
+    guard let fileURL = Bundle.main.url(forResource: "sample", 
+                                        withExtension: "mp4") else { fatalError() }
+    let video = try Data(contentsOf: fileURL)
+    let prompt = "What's in this video?"
+    let videoContent = ModelContent.Part.data(mimetype: "video/mp4", video)
+
+    // To generate text output, call generateContent and pass in the prompt
+    let response = try await model.generateContent(videoContent, prompt)
+    if let text = response.text {
+      print(text)
+    }
+    // [END text_video_prompt]
+  }
+
+  func textAndVideoPromptStreaming() async throws {
+    // [START text_video_prompt_streaming]
+    guard let fileURL = Bundle.main.url(forResource: "sample",
+                                        withExtension: "mp4") else { fatalError() }
+    let video = try Data(contentsOf: fileURL)
+    let prompt = "What's in this video?"
+    let videoContent = ModelContent.Part.data(mimetype: "video/mp4", video)
+
+    // To stream generated text output, call generateContentStream and pass in the prompt
+    let contentStream = model.generateContentStream(videoContent, prompt)
+    for try await chunk in contentStream {
+      if let text = chunk.text {
+        print(text)
+      }
+    }
+    // [END text_video_prompt_streaming]
+  }
+
+  func chatStreaming() async throws {
+    // [START chat_streaming]
+    // Optionally specify existing chat history
+    let history = [
+      ModelContent(role: "user", parts: "Hello, I have 2 dogs in my house."),
+      ModelContent(role: "model", parts: "Great to meet you. What would you like to know?"),
+    ]
+
+    // Initialize the chat with optional chat history
+    let chat = model.startChat(history: history)
+
+    // To stream generated text output, call sendMessageStream and pass in the message
+    let contentStream = chat.sendMessageStream("How many paws are in my house?")
+    for try await chunk in contentStream {
+      if let text = chunk.text {
+        print(text)
+      }
+    }
+    // [END chat_streaming]
+  }
+
+  func chat() async throws {
+    // [START chat]
+    // Optionally specify existing chat history
+    let history = [
+      ModelContent(role: "user", parts: "Hello, I have 2 dogs in my house."),
+      ModelContent(role: "model", parts: "Great to meet you. What would you like to know?"),
+    ]
+
+    // Initialize the chat with optional chat history
+    let chat = model.startChat(history: history)
+
+    // To generate text output, call sendMessage and pass in the message
+    let response = try await chat.sendMessage("How many paws are in my house?")
+    if let text = response.text {
+      print(text)
+    }
+    // [END chat]
+  }
+
+  func countTokensText() async throws {
```

Comment:
question on these count tokens and billable chars snippets... can they be embedded inside each of the main snippets above?  With region tags added?

On the [count tokens docs page](https://firebase.google.com/docs/vertex-ai/count-tokens?platform=ios), I'd like to be able to show count tokens in the context of the other code.  And using special includecode syntax, there's a way to hide specific parts of a code snippet.  So, I can hide the count tokens bit in the main pages of the guides and only expose it in the count tokens page.

Reply:
I don't think there's a way to nest snippets inside the exclude syntax, but you could certainly put them side by side.